### ISSUE TITLE
`MapboxDefaultPopup` HTML doesn't react to changes

### ISSUE THREAD

**Issue by aliraza944** [AlexLavoie42/Nuxt-Mapbox#37]
# My Problem 
I have a `MapboxLayer` of some geo JSON data containing properties that I want to show on the `MapboxDefaultPopup `.  I have added an event listener called `Mouseover` on the layer. As the mouse moves from one polygon to another I want to show the properties on the popup of that polygon on which the mouse is hovering. <br />
It only shows the default popup with just one polygon and does not display for the other.

# My Code
Here is the  `MapboxLayer` that I have implemented
```javascript
<MapboxLayer
      :layer="{
        source: 'data',
        id: 'geojson-layer',
        type: 'line',
        paint: {
          'line-color': '#0052cc',
          'line-width': 2,
        },
      }"
      @mouseover="(e) => handleMouseMove(e)"
    />
```
The popup component has dynamic lat langs based on the mouse position

```javascript
<MapboxDefaultPopup
        popup-id="popup1"
        :lnglat="hoverInfo.lngLat"
        :offset="[0, -10]"
        :options="{
          closeOnClick: false,
        }"
      >
        <span class="test">Hello World! {{ hoverInfo.lngLat[0] }}</span>
      </MapboxDefaultPopup>

```

My handleMouseMove function just sets the **lat langs** for the popup
``` javascript
const handleMouseMove = (e) => {
 // hoverInfo.value = null;
  hoverInfo.value = {
    lngLat: [e.lngLat.lng, e.lngLat.lat],
  };
};

```
 # Desired Outcome
I want to show the popup dynamically for the polygon on which the mouse pointer is.

**Comment by AlexLavoie42** [comment]
Please leave a minimum reproduction using [the playground](https://stackblitz.com/github/AlexLavoie42/Nuxt-Mapbox?file=playground%2Fpages%2Findex.vue)

**Comment by aliraza944** [comment]
> Please leave a minimum reproduction using [the playground](https://stackblitz.com/github/AlexLavoie42/Nuxt-Mapbox?file=playground%2Fpages%2Findex.vue)

I have added my code to the playground. You can clearly see that it does not changes the `latLang` I am trying to print. Also the popup dose not shift from the one polygon to another very often.

**Comment by AlexLavoie42** [comment]
Please save a copy and provide a link. Any changes you make do not reflect on the original playground

**Comment by aliraza944** [comment]
> Please save a copy and provide a link. Any changes you make do not reflect on the original playground

Sorry forgot here is the link [https://github.com/AlexLavoie42/Nuxt-Mapbox/issues/37#issuecomment-1581248504](url)

**Comment by aliraza944** [comment]
The internal `$mapboxPopupInstances` is returning **undefined**.  Can you please share an example with the `defineMapboxPopup` cause I don't where to set it templateRef. I couldn't quite understand it from the DOCS

**Comment by AlexLavoie42** [comment]
No the ref will be assigned to whatever HTML you want to add to the popup. If you are using `defineMapboxPopup`, you can no longer use the default `MapboxDefaultPopup` component, you have to make your own component.

I highly recommend using the default popup instead and accessing the instance with `$mapboxPopupInstances`. Make sure you are indexing the correct popup ID, and you access the popup instance after the map load (`useMapbox` callback should work for this).

**Comment by AlexLavoie42** [comment]
Fixed in #45
